cascade-merge/action.yml

```diff
@@ -0,0 +1,162 @@
+# more info on https://ledgerhq.atlassian.net/l/cp/Ar8F1nmM
+name: 'Auto Cascade Merge'
+description: 'Automatically cascade merge changes from one branch to others.'
+branding:
+  icon: 'git-pull-request'
+  color: 'purple'
+
+inputs:
+  gh-token:
+    description: 'pass GITHUB_TOKEN'
+    required: true
+  default-branch:
+    description: 'Default branch name (e.g., main or master)'
+    required: true
+    default: 'main'
+  auto-merge:
+    description: 'Whether to auto-merge the pull requests or not. true/false'
+    required: true
+    default: 'false'
+
+runs:
+  using: "composite"
+
+  steps:
+    - name: Checkout code
+      uses: actions/checkout@v3
+      with:
+        fetch-depth: 0  # fetch all history so we can push to other branches
+
+    - name: Configure Git
+      run: |
+        git config user.name "GitHub Actions"
+        git config user.email "github-actions@github.com"
+      shell: bash
+
+    - name: Create PR
+      id: create_prs
+      run: |
+        # Enables the shell to print all executed commands and exit if any command fails
+        # set -ex
+        
+        # A function to compare semantic version numbers. Echoes 1 if the first is greater, 2 if the second is greater, 0 if they are equal
+        compare_versions() {
+            IFS='.' read -ra VERSION1 <<< "$1"
+            IFS='.' read -ra VERSION2 <<< "$2"
+        
+            for i in {0..2}; do
+                PART1=${VERSION1[i]:-0}
+                PART2=${VERSION2[i]:-0}
+                if (( PART1 > PART2 )); then
+                    echo 1
+                    return
+                elif (( PART1 < PART2 )); then
+                    echo 2
+                    return
+                fi
+            done
+            echo 0
+        }
+        
+        # beautify the message by coloring it to a green color
+        function echo_green {
+            echo -e "\e[32m$1\e[0m"
+        }
+        
+        PREFIX="auto-cascade-merge"
+        ORIGINAL_BRANCH="${GITHUB_REF#refs/heads/}"
+        # backport target branch pattern is harcoded here as `release/`
+        BACKPORT_TARGET_BRANCH_PATTERN="release/"
+        # Remove the branch pattern prefix from the original branch name to get the original version
+        ORIGINAL_VERSION="${ORIGINAL_BRANCH#"$BACKPORT_TARGET_BRANCH_PATTERN"}"
```

Comment:
I don't understand what's this supposed to be ? :thinking: 
I tested quickly locally and it's empty, but maybe something is missing in my env

Reply:
`ORIGINAL_VERSION="${ORIGINAL_BRANCH#"$BACKPORT_TARGET_BRANCH_PATTERN"}`
is the version of the current branch 
I pushed a commit to release/7.12
`${ORIGINAL_BRANCH#"$BACKPORT_TARGET_BRANCH_PATTERN"}` => `release/7.12#release/` => 7.12